src/test/java/fi/helsinki/cs/tmc/core/domain/CourseTest.java

```diff
@@ -103,4 +103,40 @@ public void hashCodeGeneratedFromId() {
         c2.setId(1);
         assertTrue(courses.contains(c2));
     }
+
+    @Test
+    public void courseExercisesAreDividedIntoThemesCorrectly() {
+        exercises = new ArrayList<>();
+        exercises.add(new Exercise("viikko1-testi"));
+        exercises.add(new Exercise("viikko2-testi"));
+        exercises.add(new Exercise("viikko3-testi"));
+        course.setExercises(exercises);
+        course.generateThemes();
+        assertEquals(3,course.getExercises().size());
```

Comment:
You aren't really testing the new code here in any capacity beyond `generateThemes()` not producing an exception.